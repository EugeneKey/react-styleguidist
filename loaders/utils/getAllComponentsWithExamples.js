'use strict';

/**
 * Get all components that have example files.
 *
 * @param {Array} sections
 * @returns {Array}
 */
module.exports = function getAllComponentsWithExamples(sections) {
	return sections.reduce((components, section) => {
		if (section.components) {
			components = components.concat(section.components.filter(component => component.hasExamples));
		}

		if (section.sections) {
			components = components.concat(getAllComponentsWithExamples(section.sections));
		}

		return components;
	}, []);
};
