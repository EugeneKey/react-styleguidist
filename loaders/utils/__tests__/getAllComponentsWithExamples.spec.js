import getAllComponentsWithExamples from '../getAllComponentsWithExamples';

const sections = [
	{
		name: 'Readme',
		content: 'Readme.md',
		components: [],
		sections: [],
	},
	{
		name: 'Components',
		components: [
			{
				filepath: 'components/Button/Button.js',
				hasExamples: true,
			},
			{
				filepath: 'components/Icon/Icon.js',
			},
		],
		sections: [],
	},
	{
		name: 'Nesting',
		components: [],
		sections: [
			{
				name: 'Nested',
				components: [
					{
						filepath: 'components/Image/Image.js',
					},
					{
						filepath: 'components/Modal/Modal.js',
						hasExamples: true,
					},
				],
				sections: [],
			},
			{
				name: 'Nested 2',
				components: [
					{
						filepath: 'components/Avatar/Avatar.js',
					},
				],
				sections: [],
			},
		],
	},
];

it('should return only components with example file', () => {
	const result = getAllComponentsWithExamples(sections);
	expect(result).toMatchSnapshot();
});
