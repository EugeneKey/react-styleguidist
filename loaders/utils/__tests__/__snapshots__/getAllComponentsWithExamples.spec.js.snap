exports[`test should return only components with example file 1`] = `
Array [
  Object {
    "filepath": "components/Button/Button.js",
    "hasExamples": true,
  },
  Object {
    "filepath": "components/Modal/Modal.js",
    "hasExamples": true,
  },
]
`;
